In [3]:
#1
n=6
num=1
for i in range(n):
    print(num,end="")
    num+=1

123456

In [5]:
#2
n=6
for i in range(n):
    print("*",end="")


******

PROBLEM 3) Rectangle
Sample input:
4
6
Sample output
******
******
******
******

Hint: 4 and 6 means 4 lines of stars having 6 stars in each line.

In [12]:
l=4
c=6
for line in range(l):
    for star in range(c):
        print("*",end="")
    print()

******
******
******
******


PROBLEM 4) Rectangle
Sample input:
4
6
Sample output
123456
123456
123456
123456
Hint: 4 and 6 means 4 lines of numbers having 1..6 in each line.

In [2]:
l=4
c=6

for line in range(l):
    num=1
    for star in range(c):
        print(num,end="")
        num+=1
    print()

123456
123456
123456
123456


PROBLEM 5) Triangle - Left Justified
Draw right angled triangle of given height
Sample input:
4
Sample output
*
**
***
****
Hint: One loop for lines, another loop for printing i number of starts when it is line i.

In [14]:
#5
n=4
star="*"
for i in range(n):
    print(star)
    star+="*"
    

*
**
***
****


PROBLEM 6) Triangle - Left Justified
Draw right angled triangle of given height
Sample input:
4
Sample output
1
12
123
1234

In [2]:
#6
n=4                             
for i in range(n):#i=0,1,2,3
    num=1         #1,2,3,4
    for j in range(i+1): #1,2,3,4
        print(num,end="")#1,
        num+=1  #2  
    print()       

1
12
123
1234


In [4]:
#6
n=4                              
for i in range(1,n+1):#i=0,1,2,3
    num=1         #1,2,3,4
    for j in range(1,i+1): #1,2,3,4
        print(j,end="")#1,
        num+=1  #2  
    print()       

1
12
123
1234


PROBLEM 7) Triangle - Right Justified
Draw right angled triangle of given height
Sample input:
4
Sample output
   *
  **
 ***
****
Hint: Count and print appropriate number of spaces in front of stars. Notice that there
is one less space and one more star in each line.

In [15]:
#7
n=4
count=n-1
for i in range(n):
        for s in range(count):
            print(end=" ")
        count=count-1
        for j in range(i+1):
            print("*",end="")
        print()

   *
  **
 ***
****


PROBLEM 8) Triangle - Right Justified
Draw right angled triangle of given height
Sample input:
4
Sample output
   1
  12
 123
1234

In [20]:
#8
n=4
space=n-1

for i in range(n):
    for s in range(space):
        print(" ",end="")
    space=space-1
    num=1                    
    for j in range(i+1):
        print(num,end="")
        num+=1
    print()

   1
  12
 123
1234


PROBLEM 9) Triangle - Isosceles
Draw triangle of given height
Sample input 1:
3
Sample output 1:
  *
 ***
*****
Sample input 2:
4
Sample output 2:
   *
  ***
 *****
*******

In [6]:
#9
n=4
s=n-1
star="*"
for i in range(n):
    for j in range(s):
        print(end=" ")
    s=s-1
    print(star,end="")
    star+="**"
    print()

   *
  ***
 *****
*******


PROBLEM 10) Triangle - Isosceles
Draw triangle of given height
Sample input 1:
3
Sample output 1:
  1
 123
12345
Sample input 2:
4
Sample output 2:
   1
  123
 12345
1234567

In [28]:
n=4
t=1
space=n-1
for i in range(n):
    for j in range(space):
        print(end=" ")
    space-=1
    for k in range(1,t+1):
        print(k,end="")
    t+=2
    print()

   1
  123
 12345
1234567


PROBLEM 11) Triangle - Right Justified
Draw right angled triangle of given height
Sample input:
4
Sample output
   4
  34
 234
1234

In [4]:
n=4
space=n-1
                                          #baki

for i in range(n):
    x=1
    for j in range(space):
        print(end=" ")
        x+=1
    space-=1
    
    for k in range(i+1):
        
        print(x,end="")
        x+=1                  
    
    print()    

   4
  34
 234
1234


PROBLEM 12) Rhombus
Just draw the image of the above triangle once. And then, the opposite, once.
Sample input:
3
Sample output
  *
 ***
*****
 ***
  *

In [15]:
n=8
c=n
space=n-1
for row in range(1,n*2):
    for colum in range(1,c+1):
        if colum<=space:
            print(" ",end="")
        else:
            print("*",end="")
    
    if row <n:
        c+=1
        space-=1
    else:
        c-=1
        space+=1
    
    print()
            

       *
      ***
     *****
    *******
   *********
  ***********
 *************
***************
 *************
  ***********
   *********
    *******
     *****
      ***
       *


PROBLEM 13) Rhombus
Just draw the image of the above triangle once. And then, the opposite, once.
Sample input:
3
Sample output
  1
 123
12345
 123
  1

In [22]:
n=5
c=n  #3
space=n-1   #1
for row in range(1,n*2):
    num=1
    for colum in range(1,c+1):
        if colum<=space:
            print(" ",end="")
        else:
            print(num,end="")
            num+=1
    print()
    if row<n:
        c+=1
        space-=1
    else:
        c-=1
        space+=1
        

    1
   123
  12345
 1234567
123456789
 1234567
  12345
   123
    1


PROBLEM 14) Hollow Rectangle
Display a rectangle of given length and width.
Sample input:
4
5
Sample output
*****
*   *
*   *
*****
Hint 1: Print the character space (‘ ‘) in the middle.
Hint 2: You can re-use your solution to PROBLEM 2 and use if condition to selectively print first and last star of
each line and all stars of first and last line.

In [72]:
l=5
w=9
for row in range(1,l+1):
    for colum in range(1,w+1):
        if row==1 or row==l:
            print("*",end="")
            
        else:
            if colum==1 or colum==w:
                print("*",end="")
            else:
                print(" ",end='')
    print()

*********
*       *
*       *
*       *
*********


PROBLEM 15) Hollow Rectangle
Display a rectangle of given length and width.
Sample input:
4
5
Sample output
12345
1   5
1   5
12345

In [73]:
l=4
w=5
for row in range(1,l+1):
    num=1
    for colum in range(1,w+1):
        if row==1 or row==l:
            print(num,end="")
            num+=1
        else:
            if colum==1 or colum==w:
                print(num,end="")
                num+=1
            else:
                print(" ",end="")
                num+=1
    print()

12345
1   5
1   5
12345


PROBLEM 16) Hollow Triangle - Left Justified
Draw right angled triangle of given height
Sample input:
5
Sample output
*
**
* *
*  *
*****


In [88]:
h=14
c=1
for row in range(1,h+1):   #1
    for colum in range(1,c+1): #1
        if row==h or row<3:
            print("*",end="")
        
        else: #3
            if colum==1 or colum==c:
                print("*",end="")
            else:
                print(" ",end="")
        
    c+=1
    print()

*
**
* *
*  *
*   *
*    *
*     *
*      *
*       *
*        *
*         *
*          *
*           *
**************


PROBLEM 17) Hollow Triangle - Left Justified
Draw right angled triangle of given height
Sample input:
5
Sample output
1
12
1 3
1  4
12345

In [99]:
h=5
c=1
for row in range(1,h+1):
    num=1
    for colum in range(1,c+1):
        if row<3 or row==h:
            print(num,end="")
            num+=1
        else:
            if colum==1 or colum==c:
                print(num,end="")
                num+=1
            else:
                print(" ",end="")
                num+=1
    print()
    c+=1

1
12
1 3
1  4
12345


PROBLEM 18) Hollow Triangle
Draw right angled triangle of given height
Sample input:
5
Sample output
    *
   **
  * *
 *  *
*****

In [25]:
h=5
c=h-1
x=1
for row in range(1,h+1):
    for colum in range(1,h-row+1):
        print(" ",end="")
    
    for colum in range(1,row+1):
        if row==1 or row==2 or row==h:
            print("*",end="")
        else:
            if colum==1 or colum==row:
                print("*",end="")
            else:
                print(" ",end="")
            
            
    
    x+=1
    c-=1
    print()

    *
   **
  * *
 *  *
*****


PROBLEM 19) Hollow Triangle - Right Justified
Draw right angled triangle of given height
Sample input:
5
Sample output
    5
   45
  3 5
 2  5
12345

In [33]:
n=5
c=n-1
x=1
for row in range(1,n+1):
    num=1
    for colum in range(1,c+1):
        print("*",end="")
        num+=1
    for colum in range(1,x+1):
        if row==1 or row==2 or row==n:
            print(num,end="")
            num+=1
        else:
            if colum==1 or colum==x:
                print(num,end="")
                num+=1
            else:
                print(" ",end="")
                num+=1
    x+=1
    print()    
    c-=1
    

****5
***45
**3 5
*2  5
12345


PROBLEM 20) Hollow Triangle
Draw triangle of given height
Sample input 1:
3
Sample output 1:
  *
 * *
*****
Sample input 2:
4
Sample output 2:
   *
  * *
 *   *
*******

In [49]:
h=4
c=h-1
x=1
for row in range(1,h+1):
    
    for colum in range(1,c+1):
        print(" ",end="")
    if row!=h:
        
        for colum in range(1,x+1):
            if colum==1 or colum==x:
                print("*",end="")
            else:
                print(" ",end="")
    else:
        for colum in range(1,x+1):
            
            print("*",end="")
        
    
    print()
    c-=1
    x+=2

   *
  * *
 *   *
*******


PROBLEM 21) Hollow Triangle – Isosceles
Draw triangle of given height
Sample input 1:
3
Sample output 1:
  1
 1 3
12345
Sample input 2:
4
Sample output 2:
   1
  1 3
 1   5
1234567

In [4]:
h=4
c=h-1
x=1
for row in range(1,h+1):
    num=1
    for colum in range(1,c+1):
        print(" ",end="")
    for colum in range(1,x+1):
        if row==1 or row==h:
            print(num,end="")
            num+=1
        else:
            if colum==1 or colum==x:
                print(num,end="")
                num+=1
            else:
                print(" ",end="")
                num+=1
        
    print()
    c-=1
    x+=2

   1
  1 3
 1   5
1234567


PROBLEM 22) Hollow Rhombus
Just draw the image of the above triangle once. And then, the opposite, once.
Sample input:
3
Sample output
  *
 * *
*   *
 * *
  *

In [36]:
n=7
c=n-1
x=1
s=1
p=n*2-3
for row in range(1,n*2):
    if row<=n:
        for colum in range(1,c+1):
            print(" ",end="")
        for colum in range(1,x+1):
            if colum==1 or colum==x:
                print("*",end="")
            else:
                print(" ",end="")
        
        print()
        c-=1
        x+=2
    else:
        for colum in range(1,s+1):
            print(" ",end="")
        for colum in range(1,p+1):
            if colum==1 or colum==p:
                print("*",end="")
            else:
                print(" ",end="")
        
        
        print()
        s+=1 
        p-=2

      *
     * *
    *   *
   *     *
  *       *
 *         *
*           *
 *         *
  *       *
   *     *
    *   *
     * *
      *


PROBLEM 24) Palindrome
Sample input:
5
Sample output
123454321

In [8]:
n=6
for i in range(1,n+1):
    print(i,end="")
for e in range((n-1),0,-1):
    print(e,end="")

12345654321

PROBLEM 25) Palindromic Triangle
Sample input:
5
Sample output
    1
   121
  12321
 1234321
123454321

In [2]:
n=5
p=n-1
for row in range(1,n+1):
    for s in range(p,0,-1):
        print(" ",end="")
    for t in range(1,row+1):
        print(t,end="")
    for y in range(row-1,0,-1):
        print(y,end="")
    p-=1
    print()

    1
   121
  12321
 1234321
123454321
